### ISSUE TITLE
⌨ Process navigation not announced to ATs

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#9728]
## ⌨ Accessibility Issue

On page pages, we set focus to the `h1` so it is announced to ATs and orients them on context changes. However, since the `PoolLayout` never unmounts and is consistent across navigations, it is not clear to AT users when we navigate that section.

### WCAG 2.1 Criteria

- [4.1.3: Status Messages (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)
- [Changes of Context - Single Page Applications](https://stevenmouret.github.io/web-accessibility-guidelines/accessibility/scripts/changes-of-context.html#:~:text=accessible%20name.-,Single%20page%20application,-In%20a%20%E2%80%9CSingle)

## 🦋 Expected Behaviour

Screen reader users should be notified of a page/content change.
`
## 📋 Steps to Reproduce

1. Load up any screen reader you have access to (this affects all)
2. Login as admin `admin@test.com`
3. Navigate to a specific process
4. Navigate using the tabs in the hero
5. Notice there is no announcement of the page change

## 📸 Screenshot

![Screenshot 2024-03-13 080814](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/8c5afadb-1cd7-4907-a008-a53e030a8ec1)

## 🙋‍♀️ Proposed Solution

~This will need some help from designers. I think we need some sub title that we can focus in the event that the actual title does not change. I think we could use some attribute like `data-a11y-status` on that subtitle and use it in the hero's focus to move it there instead of the `h1` if it exists.~

Another option is maybe trying to use out `useAnnouncer` hook and attempt to do some announcements from the layout when the current page changes so we don't need to change the design at all?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] When navigating the process admin pages all navigations (changes in context) are announced to ATs

**Comment by gobyrne** [comment]
Jerbo, could you take a look at what Eric is proposing please?

**Comment by tristan-orourke** [comment]
There seems to be a preference for solution 2, which seems easier to implement for now.
